tests/cases/fourslash/formatAddSemicolons1.ts

```diff
@@ -0,0 +1,49 @@
+/// <reference path="fourslash.ts" />
+
+////console.log(1)
+////console.log(2)
+////const x = function() { }
+////for (let i = 0; i < 1; i++) {
+////    1
+////    2
+////}
+////do { } while (false) console.log(3)
+////function f() { }
+////class C {
+////    ["one"] = {}
+////    ["two"]
```

Comment:
1. what happens with existing semicolons?
2. why aren't there semicolons here in the original source? are you modelling erroneous code?
3. if so, can you add

```ts
;['three'] = {}
;['four']
```

to show what happens there?

Reply:
> why aren't there semicolons here in the original source? are you modelling erroneous code?

Valid, but stupidly formatted code. This parses as `["one"] = {}["two"]`, so the test ensures that we don’t insert a semicolon that would change how the input parses.

> if so, can you add

Yeah, sure. I can tell you that the formatter won’t even consider it, as a potential spot for insertion, because it processes pairs of tokens excluding trivia, so it will see a declaration that already ends in a semicolon. But it’s nice to add realistic user code as tests along with these bananas edge cases.

Reply:
> erroneous code

> stupidly formatted code

You say [tʰəme<sup>ɪ</sup>ɾo<sup>ʊ</sup>], I say [tʰəmato<sup>ʊ</sup>]!